datapipelines/src/test/java/io/customer/datapipelines/DataPipelinesInteractionTests.kt

```diff
@@ -72,6 +72,29 @@ class DataPipelinesInteractionTests : JUnitTest() {
         identifyEvent.traits.shouldBeEqualTo(emptyJsonObject)
     }
 
+    @Test
+    fun identify_givenProfileAttributesAndNoIdentifier_expectSetNewProfileWithoutAttributes() {
+        val givenIdentifier = String.random
+
+        analytics.userId().shouldBeNull()
+        every { globalPreferenceStore.getDeviceToken() } returns String.random
+
+        sdkInstance.profileAttributes = mapOf("first_name" to "Dana", "ageInYears" to 30)
+        analytics.userId() shouldBe ""
```

Comment:
I might be missing something but isn't setting the ID to empty string is what we are trying to fix?

Reply:
We are handling the case where it would set the ID to empty, we can't stop from setting the id to empty. When you set profile attributes before identifying, it sets identifier to empty string in the Segment SDK. 

So this test verifies that, when we set profile attributes before, we assert identifier is being set to an empty string and then when we identify it later on, it doesn't effect the flow because our code is handling the empty string along with `null`.

Reply:
So basically we want `currentlyIdentifiedProfile` to be `null` instead of empty string and that fixes the issue?

Reply:
Yes, because that's the default case. When there is no user identified, identifier should be null. So now if it's empty (edge case) we also consider it as null.